src/vinyldns/batch_change.py

```diff
@@ -157,12 +163,16 @@ def __init__(self, zone_id, zone_name, record_name, input_name, type, status,
 
     @staticmethod
     def from_dict(d):
+        record = d.get('record')
```

Comment:
We can't just outright do `rdata_converters[d['type']](d['record'])` since `record` is optional. `rdata_converters[d['type']](d.get('record'))` and `map(lambda r: rdata_converters[d['type']](r), d.get('record'))` both fail when `record` is missing (i.e. `None`).

Reply:
Circumvented this with `map_option`.